app/views/assignments/_timed_due_info.html.erb

```diff
@@ -0,0 +1,40 @@
+<% due_date = @assignment.section_due_date(@current_user&.section)
+   start_time = @assignment.section_start_time(@current_user&.section) %>
+<% duration = @grouping.nil? ?
+                @assignment.assignment_properties.adjusted_duration(due_date, [Time.current, start_time].max) :
+                @grouping.adjusted_duration %>
+<h3><%= I18n.t('activerecord.models.timed_assignment.one') %></h3>
+<p>
+  <strong><%= I18n.t('assignments.timed.start_time_instructions') %></strong>
+  <%= @assignment.section_start_time(@current_user&.section).strftime('%Y-%m-%d %l:%M %p') %>
+<p>
+  <strong><%= I18n.t('assignments.timed.end_time_instructions') %></strong>
+  <%= due_date.strftime('%Y-%m-%d %l:%M %p') %>
+</p>
+<% if @grouping&.start_time.nil? %>
+  <% if Time.current <= due_date && Time.current >= start_time %>
+    <p>
+      <% parts = AssignmentProperties.duration_parts(duration)
+         duration_string = ["#{parts[:hours]} #{I18n.t('durations.hours', count: parts[:hours])}",
+                            "#{parts[:minutes]} #{I18n.t('durations.minutes', count: parts[:minutes])}"].join(', ')
+      %>
+      <%= I18n.t('assignments.timed.before_start_instructions', duration_string: duration_string)%>
+    </p>
+    <%= form_with url: start_timed_assignment_assignment_path(@assignment.id), method: 'put', local: true do |f| %>
+      <%= f.hidden_field :grouping_id, value: @grouping.id %>
```

Comment:
There's something to investigate here. The condition on line 14 allows for the case that `@grouping` is `nil`, but here this isn't allowed. Perhaps this form needs to be guarded by an `if @grouping.nil?` and separate instructions given to students to form a (valid?) grouping first.

Reply:
Oh, good catch. I rewrote this view so many times now, I definitely just missed this.